src/app/services/auth.service.ts

```diff
@@ -1,14 +1,31 @@
 import { Injectable } from '@angular/core';
 import { LocalStorageService } from './local-storage.service';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthService {
-  constructor(private localStorageService: LocalStorageService) {}
+  private authUrl = 'http://localhost:3004/auth';
 
-  login(user: any): boolean {
-    return this.localStorageService.setItemInLocalStorage('user', user);
+  constructor(
+    private localStorageService: LocalStorageService,
+    private http: HttpClient
+  ) {}
+
+  login(user: any, successCallback, errorCallback) {
+    return this.http
```

Comment:
This returns Subscription that does not make much sense actually.
I would prefer to return an Observable here and you should find the suitable operator for doing side effects like setting token to localStorage.